src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -5665,9 +5685,65 @@ bool StorageReplicatedMergeTree::waitForShrinkingQueueSize(size_t queue_size, UI
     return true;
 }
 
+bool StorageReplicatedMergeTree::dropPart(
+    zkutil::ZooKeeperPtr & zookeeper, String part_name, LogEntry & entry, bool detach)
+{
+    LOG_TRACE(log, "Will try to insert a log entry to DROP_RANGE for part: " + part_name);
+
+    auto part_info = MergeTreePartInfo::fromPartName(part_name, format_version);
+
+    while (true)
+    {
+        ReplicatedMergeTreeMergePredicate merge_pred = queue.getMergePredicate(zookeeper);
+
+        /// TODO(nv) It is possible that part does not exist on replica which executes this statement.
```

Comment:
I think this TODO item is an edge case which can be handled separately in another PR to keep this one simpler and merge it quicker.

Reply:
It should exist in the queue's `virtual_parts` or at least in zookeeper?